README.md

```diff
@@ -70,8 +151,16 @@ Marsha is compiled by an LLM into tested software that meets the requirements de
 ## Roadmap
 
 - Improve average accuracy for our test bed above 90%
-- Support for endpoints, visualizations, and data storage
-- Syntax highlighting
+- Support for visualizations and data storage (geek mode: handle side-effect logic better in general)
+- Syntax highlighting (vim, vscode, etc)
 - Support for different types of LLM
-- Being able to bootstrap the Marsha compiler with a Marsha program
-- More target languages other than Python
\ No newline at end of file
+- Bootstrap the Marsha compiler with a Marsha program
+- More target languages other than Python
+- A module system
+- Edits to Marsha mutating existing Python code instead of regenerating
+- "Decompiler" from source code into Marsha syntax
+- "Debugger" meta mode to take existing Marsha definition and an example of an unexpected failure and recommend what to update with the Marsha definition.
+- Optmization "levels" (spend more time on more iterations with the LLM improving performance, security, etc)
+- Marsha GUI mode: visual editor baked into the compiler (eventually with the decompiler/debugger/etc features), and able to generate a GUI wrapper for generated code, enabling end-to-end non-terminal usage
+- Better support for a mixed environment (Marsha functions can be used by Python, but how to get Marsha to use hand-written Python functions)
```

Comment:
Yes, sorta. I don't know if this should be opt-in or opt-out, but having Marsha handle a whole "tree" of marsha files within a project written primarily in another language would be useful.

This would require stuff like:

* Understanding project structure in every target language (to be able to find all of the other code in the project, to know where to put the test files, etc)
* Having the "edit existing code" working, since every Marsha change producing a full rewrite of the code would make code review difficult/impossible
* Perhaps some special syntax to specify what code a Marsha function should depend on, rather than leaving it up to the whims of the LLM (we can't automatically confirm usage of it, but we should be able to automatically confirm *import* of the desired functionality, which will *usually* convince the LLM to use it).
* Probably some other things I haven't thought of, yet.